app/src/main/java/com/mapbox/maps/testapp/examples/markersandcallouts/ViewAnnotationAnimationActivity.kt

```diff
@@ -0,0 +1,162 @@
+
+import android.animation.Animator
+import android.animation.AnimatorListenerAdapter
+import android.animation.TypeEvaluator
+import android.animation.ValueAnimator
+import android.animation.ValueAnimator.AnimatorUpdateListener
+import android.graphics.Color
+import android.os.Bundle
+import android.view.View
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import com.mapbox.geojson.FeatureCollection
+import com.mapbox.geojson.LineString
+import com.mapbox.geojson.Point
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.MapView
+import com.mapbox.maps.Style
+import com.mapbox.maps.ViewAnnotationOptions
+import com.mapbox.maps.extension.style.layers.generated.lineLayer
+import com.mapbox.maps.extension.style.sources.generated.geoJsonSource
+import com.mapbox.maps.extension.style.style
+import com.mapbox.maps.testapp.R
+import com.mapbox.maps.testapp.databinding.ItemCalloutViewBinding
+import com.mapbox.maps.testapp.examples.annotation.AnnotationUtils
+import com.mapbox.maps.viewannotation.ViewAnnotationManager
+import com.mapbox.turf.TurfConstants.UNIT_DEFAULT
+import com.mapbox.turf.TurfMeasurement
+import kotlin.math.roundToLong
+
+/**
+ * Example how to animate a view annotations on top of a route.
+ */
+class ViewAnnotationAnimationActivity : AppCompatActivity() {
+
+  private lateinit var viewAnnotationManager: ViewAnnotationManager
+  private lateinit var annotationView: View
+  private lateinit var textView: TextView
+  private var routeCoordinateList = mutableListOf<Point>()
+  private var currentRouteCoordinate = 0
+  private var totalLength = 0.0
+  private var currentAnimator: Animator? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    val mapView = MapView(this)
+    setContentView(mapView)
+
+    // load feature collection from assets file
+    val featureCollection = FeatureCollection.fromJson(
+      AnnotationUtils.loadStringFromAssets(
+        this, ROUTE_FILE_NAME
+      )!!
+    )
+
+    // calculate the route length, get coordinates from route geometry
+    val lineString = featureCollection.features()!![0].geometry() as LineString
+    routeCoordinateList = lineString.coordinates()
+    totalLength = TurfMeasurement.length(lineString, UNIT_DEFAULT)
+
+    // initialize the mapview
+    mapView.getMapboxMap().apply {
+      loadStyle(
+        style(Style.LIGHT) {
+          // source for displaying the route
+          +geoJsonSource(SOURCE_ID) {
+            featureCollection(featureCollection)
+          }
+          // layer for displaying the route
+          +lineLayer(LINE_ID, SOURCE_ID) {
+            lineColor(Color.parseColor(COLOR_PINK_HEX))
+            lineWidth(4.0)
+          }
+        }
+      ) {
+        // center camera around SF airport
+        setCamera(
+          CameraOptions.Builder()
+            .center(Point.fromLngLat(-122.3915, 37.6177))
```

Comment:
nit: move to constant, one idea, maybe we should have some global util constants to host some frequently-used geo locations?